### PR TITLE
feat(components/tabbar): rename selected to selectedKey

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

The `selected` is reserved which is causing issues with `ngReact`

**What is the chosen solution to this problem?**

Rename `selected` to `selectedKey`.

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**

### CODE REVIEW

**Comment by jsomsanith** [comment]
I don't understand the need.
The previous version ensure that only 1 tab is selected. With this change, it is possible to have multiple tab selected, and from 2 possible inputs.

File: output/components.eslint.txt [hunk 1]
```diff
@@ -23,11 +23,14 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   1:1  error  '@storybook/react' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   4:2  error  Unexpected require()                                                                    global-require
 
+/home/travis/build/Talend/ui/packages/components/src/TabBar/TabBar.component.js
+  8:31  error  'isSelected' is missing in props validation  react/prop-types
```
Comment (jmfrancois) [inline output/components.eslint.txt]: please fix

File: packages/components/src/TabBar/TabBar.component.js [hunk 2]
```diff
@@ -22,16 +22,20 @@ Tab.propTypes = {
 		key: PropTypes.string.isRequired,
 		label: PropTypes.string.isRequired,
 	}).isRequired,
-	isSelected: PropTypes.bool,
```
Comment (jmfrancois) [inline packages/components/src/TabBar/TabBar.component.js]: this one should stay ?

File: packages/components/src/TabBar/TabBar.test.js [hunk 3]
```diff
@@ -22,19 +22,48 @@ const props = {
 		},
 	],
 	onSelect: onClick,
-	selected: '2',
```
Comment (jmfrancois) [inline packages/components/src/TabBar/TabBar.test.js]: why did you remove and not rename ?

File: packages/components/src/Layout/__snapshots__/Layout.test.js.snap [hunk 4]
```diff
@@ -591,7 +591,7 @@ exports[`Layout should render layout with TabBar component 1`] = `
               </button>
             </li>
             <li
-              className="tc-tab-bar-action active"
+              className="tc-tab-bar-action"
```
Comment (jmfrancois) [inline packages/components/src/Layout/__snapshots__/Layout.test.js.snap]: please take a look why theses snapshots has change here
